stats/aggregated_stats.go

```diff
@@ -23,12 +23,9 @@ func convertToAggregatedStats(id string, containerIds map[string]string, resourc
 		aggregatedStats := []AggregatedStat{}
 		for _, stat := range stats {
 			if len(stat.Stats) > i {
-				if resourceType == "container" && id == "" {
-					id = stat.Name
-				}
-				statPoint := convertCadvisorStatToAggregatedStat(id, containerIds, resourceType, memLimit, stat.Stats[i])
+				statPoint := convertCadvisorStatToAggregatedStat(id, stat.Name, containerIds, resourceType, memLimit, stat.Stats[i])
 				if statPoint.Id == "" {
-					return totalAggregatedStats
+					continue
```

Comment:
returning the complete stats when one of the Ids is not found would result in getting no stats at all, as there will be an empty id on every loop from the rancher-agent container.